packages/dds/tree/src/test/feature-libraries/chunked-forest/chunkEncodingEndToEnd.ts

```diff
@@ -0,0 +1,159 @@
+import { SummarizeType, TestTreeProvider, jsonSequenceRootSchema } from "../../utils";
+import { noopValidator } from "../../../codec";
+import { brand } from "../../../util";
+import {
+	Chunker,
+	defaultChunkPolicy,
+	tryShapeFromSchema,
+	// eslint-disable-next-line import/no-internal-modules
+} from "../../../feature-libraries/chunked-forest/chunkTree";
+// eslint-disable-next-line import/no-internal-modules
+import { decode } from "../../../feature-libraries/chunked-forest/codec/chunkDecoding";
+
+// TODO: Currently we split up a uniform chunk into several individual basicChunks for each node during op creation.
+// Therefore, there is currently no way for us to retrieve a uniform chunk from the tree for us to make the proper checks,
+// and the tests are expected to fail. The tests can be unskipped once uniform chunks can be inserted into the tree.
+describe.skip("End to End chunked encoding", () => {
+	it(`insert op values are correct, and shares reference with the original chunk.`, async () => {
+		const treeSchema = new TreeStoredSchemaRepository(intoStoredSchema(jsonSequenceRootSchema));
+		const chunker = new Chunker(
+			treeSchema,
+			defaultSchemaPolicy,
+			Number.POSITIVE_INFINITY,
+			Number.POSITIVE_INFINITY,
+			defaultChunkPolicy.uniformChunkNodeCount,
+			tryShapeFromSchema,
+		);
+
+		const forest = buildChunkedForest(chunker);
+		const numberShape = new TreeShape(leaf.number.name, true, []);
+		const chunk = new UniformChunk(numberShape.withTopLevelLength(4), [1, 2, 3, 4]);
+		assert(!chunk.isShared());
+		const chunkCursor = chunk.cursor();
+
+		// This function is declared in the test to have access to the original uniform chunk for comparison.
+		function createFlexTree(editableForest: IEditableForest, schema: FlexTreeSchema): Context {
+			const changes: ModularChangeset[] = [];
+			const changeReceiver = (change: ModularChangeset) => {
+				assert(change.builds !== undefined);
+				// TODO: This chunk is actually a BasicChunk which was split from the original UniformChunk split up.
+				// This is expected to fail currently, but should eventually pass once we have the ability to insert UniformChunk.
+				const insertedChunk = change.builds.get(undefined)?.get(0 as ChangesetLocalId);
+				assert.equal(insertedChunk, chunk);
+				assert(chunk.isShared());
+				changes.push(change);
+			};
+			const dummyEditor = new DefaultEditBuilder(
+				new DefaultChangeFamily({ jsonValidator: noopValidator }),
+				changeReceiver,
+			);
+			return getTreeContext(
+				schema,
+				editableForest,
+				dummyEditor,
+				createMockNodeKeyManager(),
+				brand(nodeKeyFieldKey),
+			);
+		}
+
+		const flexTree = createFlexTree(forest, jsonSequenceRootSchema);
+		// This conversion is currently required for inserting a uniform chunk with a topLevelLength > 1.
+		const content: ITreeCursorSynchronous[] = prepareFieldCursorForInsert(chunkCursor);
+
+		flexTree.editor
```

Comment:
this is supposed to be an end to end test, so I think it would be better to use the editing api on the flex tree, like `flexTree.editableTree.insertAt(0, chunk.cursor());` like you did in the test below. This means your test includes the logic in flex tree (which we do want to test for this, and is currently part of why it fails), and you don't need your own prepareFieldCursorForInsert

Reply:
Updated to use `insertAt(0, chunk.cursor());`